SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -40,46 +40,64 @@ FORCEINLINE UClass* ResolveClass(FString& ClassPath)
 	return Class;
 }
 
-void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
+ESchemaComponentType GetRpcType(UFunction* RemoteFunction)
 {
-	checkf(IsSupportedClass(Class), TEXT("Could not find class in schema database: %s"), *Class->GetPathName());
-
-	TSharedRef<FClassInfo> Info = ClassInfoMap.Add(Class, MakeShared<FClassInfo>());
-	Info->Class = Class;
-
-	TArray<UFunction*> RelevantClassFunctions = improbable::GetClassRPCFunctions(Class);
+	EFunctionFlags Flags = RemoteFunction->FunctionFlags;
 
-	for (UFunction* RemoteFunction : RelevantClassFunctions)
+	if (Flags & FUNC_NetMulticast)
 	{
-		ESchemaComponentType RPCType = SCHEMA_Invalid;
-		if (RemoteFunction->FunctionFlags & FUNC_NetClient)
-		{
-			RPCType = SCHEMA_ClientRPC;
-		}
-		else if (RemoteFunction->FunctionFlags & FUNC_NetServer)
+		return SCHEMA_NetMulticastRPC;
+	}
+	if (Flags & FUNC_NetCrossServer)
+	{
+		return SCHEMA_CrossServerRPC;
+	}
+	if (RemoteFunction->HasAnyFunctionFlags(FUNC_NetReliable))
+	{
+		if (Flags & FUNC_NetClient)
 		{
-			RPCType = SCHEMA_ServerRPC;
+			return SCHEMA_ClientRPC;
 		}
-		else if (RemoteFunction->FunctionFlags & FUNC_NetCrossServer)
+		if (Flags & FUNC_NetServer)
 		{
-			RPCType = SCHEMA_CrossServerRPC;
+			return SCHEMA_ServerRPC;
 		}
-		else if (RemoteFunction->FunctionFlags & FUNC_NetMulticast)
+	}
+	else
+	{
+		if (Flags & FUNC_NetClient)
 		{
-			RPCType = SCHEMA_NetMulticastRPC;
+			return SCHEMA_ClientUnreliableRPC;
 		}
-		else
+		if (Flags & FUNC_NetServer)
 		{
-			checkNoEntry();
+			return SCHEMA_ServerUnreliableRPC;
 		}
+	}
+
+	return SCHEMA_Invalid;
+}
 
-		TArray<UFunction*>& RPCArray = Info->RPCs.FindOrAdd(RPCType);
+void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
+{
+	checkf(IsSupportedClass(Class), TEXT("Could not find class in schema database: %s"), *Class->GetPathName());
 
+	TSharedRef<FClassInfo> Info = ClassInfoMap.Add(Class, MakeShared<FClassInfo>());
+	Info->Class = Class;
+
+	TArray<UFunction*> RelevantClassFunctions = improbable::GetClassRPCFunctions(Class);
+
+	for (UFunction* RemoteFunction : RelevantClassFunctions)
+	{
+		ESchemaComponentType RPCType = GetRpcType(RemoteFunction);
+		checkf(RPCType != SCHEMA_Invalid, TEXT("Could not determine RPCType for RemoteFunction: %s"), *GetPathNameSafe(RemoteFunction));
+		
 		FRPCInfo RPCInfo;
 		RPCInfo.Type = RPCType;
-		RPCInfo.Index = RPCArray.Num();
+		RPCInfo.bReliable = RemoteFunction->HasAnyFunctionFlags(FUNC_NetReliable);
+		RPCInfo.Index = Info->RPCs.Num();
```

Comment:
Can you put a comment here regarding how we can uniquely identify RPCs based on this index on both clients and servers